src/game/Tactical/OppList.cc

```diff
@@ -619,13 +619,19 @@ static void OurTeamRadiosRandomlyAbout(SOLDIERTYPE* about);
 static void OtherTeamsLookForMan(SOLDIERTYPE* pOpponent);
 
 
-void HandleSight(SOLDIERTYPE& s, SightFlags const sight_flags)
+static bool IsSoldierValidForSightings(SOLDIERTYPE const& s)
 {
 	extern INT32 iHelicopterVehicleId; 	// I don't want to include a map screen header file for this
-	if (!s.bActive)                     return;
-	if (!s.bInSector)                   return;
-	if (s.uiStatusFlags & SOLDIER_DEAD) return;
-	if (s.bAssignment == VEHICLE && s.iVehicleId == iHelicopterVehicleId) return;
+	return !(!s.bActive
+	      || !s.bInSector
+	      || s.uiStatusFlags & SOLDIER_DEAD
```

Comment:
Should we add a `|| s.bLife <= 0` as well. It should probably be covered by `s.uiStatusFlags & SOLDIER_DEAD`, but some of the conditions below checked it before (although they checked explicitly whether its not 0).

Reply:
Yes, I don't trust the game to keep these two always in sync either.